library/include/borealis/core/touch/gesture_recognizer.hpp

```diff
@@ -0,0 +1,63 @@
+/*
+    Copyright 2021 XITRIX
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#include <functional>
+
+#include <borealis/core/input.hpp>
+
+namespace brls
+{
+
+class View;
+
+// Represents current gesture state
+enum class GestureState
```

Comment:
I think the base gesture recognized interface should be in "core/gesture.hpp" since it's part of the library "core" (users can implement their own), and then every recognizer can be in core/touch/xxx